### PR TITLE
feat: remove default receipts for network components

### PR DESCRIPTION
**Pull request by Rjected**
Removes some default `Receipt` types, replacing with the associated type from `NetworkPrimitives`.

### CODE REVIEW

File: crates/net/network/src/eth_requests.rs [hunk 1]
```diff
@@ -202,7 +202,11 @@ where
             {
                 let receipt = receipts_by_block
                     .into_iter()
-                    .map(|receipt| receipt.with_bloom())
+                    .map(|receipt| {
+                        // TODO: use `into_with_bloom` on new alloy release
+                        let bloom = receipt.bloom();
+                        ReceiptWithBloom::new(receipt, bloom)
+                    })
```
Comment (mattsse) [inline crates/net/network/src/eth_requests.rs:209]: I believe `map(ReceiptWithBloom::from)` should work here

File: crates/node/builder/src/builder/mod.rs [hunk 2]
```diff
@@ -662,11 +662,8 @@ impl<Node: FullNodeTypes> BuilderContext<Node> {
                 >,
             > + Unpin
             + 'static,
-        Node::Provider: BlockReader<
-            Receipt = reth_primitives::Receipt,
-            Block = N::Block,
-            Header = N::BlockHeader,
-        >,
+        Node::Provider:
+            BlockReader<Receipt = N::Receipt, Block = N::Block, Header = N::BlockHeader>,
```
Comment (mattsse) [inline crates/node/builder/src/builder/mod.rs:666]: once merged we can add a type alias for this `BlockReaderFor<N>`

**Review by mattsse — APPROVED** [review]
nice,

ones suggestion

File: crates/net/network/src/eth_requests.rs (hunk 1 above)
Comment (Rjected) [reply crates/net/network/src/eth_requests.rs]: oh nice, yep

**Pull request merged**